cstar/tests/blueprints/fixtures.py

```diff
@@ -1,156 +1,80 @@
-                data[key] = modify_func(value)
-            else:
-                data[key] = modify_yaml(value, condition_func, modify_func, new_path)
-    elif isinstance(data, list):
-        for index, item in enumerate(data):
-            new_path = f"{path}[{index}]"
-            if condition_func(new_path, item):
-                data[index] = modify_func(item)
-            else:
-                data[index] = modify_yaml(item, condition_func, modify_func, new_path)
-    else:
-        if condition_func(path, data):
-            return modify_func(data)
-    return data
-
-
-@pytest.fixture
-def test_blueprint_as_path(test_blueprint_as_dict, tmp_path) -> Callable[[str], Path]:
-    """Given the name of a pre-defined blueprint, returns it as a (temporary) path to an on-disk file."""
-
-    def _blueprint_as_path(name: str, local: bool = False) -> Path:
-        blueprint_dict = test_blueprint_as_dict(name, local=local)
-
-        # save the blueprint to a temporary path
-        blueprint_filepath = tmp_path / "blueprint.yaml"
-        with open(blueprint_filepath, "w") as file:
-            yaml.safe_dump(blueprint_dict, file)
-
-        return blueprint_filepath
 
-    return _blueprint_as_path
+    def _template_blueprint_to_case(
+        template_blueprint_path: Path | str, strs_to_replace: dict, **kwargs
+    ) -> Case:
+        """
+        Creates a `Case` object from a modified blueprint template.
+
+        This function reads a blueprint template file, performs string replacements as specified
+        by `strs_to_replace`, saves the modified content to a temporary file within the `tmpdir`
+        provided by pytest, and then uses this temporary blueprint file to create a `Case` instance.
+        Additional keyword arguments can be passed to the `Case.from_blueprint` method.
+
+        Parameters:
+        -----------
+        template_blueprint_path (Path | str):
+           The path to the blueprint template file.
+        strs_to_replace (dict[str, str]):
+           A dictionary where keys are substrings to find in the template,
+           and values are the replacements.
+        **kwargs:
+           Additional keyword arguments passed to `Case.from_blueprint`.
+
+        Returns:
+        --------
+            Case: An instance of the `Case` class created from the modified blueprint.
+        """
+
+        template_blueprint_path = Path(template_blueprint_path)
+
+        with open(template_blueprint_path, "r") as template_file:
+            template_content = template_file.read()
+
+        modified_template_content = template_content
+        for oldstr, newstr in strs_to_replace.items():
+            modified_template_content = modified_template_content.replace(
+                oldstr, newstr
+            )
+        temp_path = tmpdir.join(template_blueprint_path.name)
+        with open(temp_path, "w") as temp_file:
+            temp_file.write(modified_template_content)
```

Comment:
Happy with this rewrite to modify the file on disk if you find that easier (the call chain is definitely simpler), the important thing is that the blueprint used is a temporary object in a temporary directory that gets cleaned up, so that each test has no side-effects on any global state. 👍